internal/provider/resource_application.go

```diff
@@ -46,17 +48,16 @@ const (
 	StorageKey          = "storage"
 
 	resourceKeyMarkdownDescription = `
-Charm resource revisions. Must evaluate to an integer.
-
-	There are a few scenarios that need to be considered:
-	* If the plan does not specify resource revision and resources are added to the plan,
-	resources with specified revisions will be attached to the application (equivalent
-	to juju attach-resource).
-	* If the plan does specify resource revisions and:
-		* If the charm revision or channel is updated, then resources get updated to the 
-		  latest revision.
-	    * If the charm revision or channel are not updated, then no changes will take 
-		  place (juju does not have an "un-attach" command for resources).
+Charm resources. Must evaluate to a string. A resource could be a resource revision number from CharmHub or a custom OCI image resource.
+There are a few scenarios that need to be considered:
+
+* Specify a resource other than the default for a charm. Note that not all charms have resources. A resource can be specified by a revision number or by URL to a OCI image repository. Resources of type 'file' can only be specified by revision number. Resources of type 'oci-image' can be specified by revision number or URL.
```

Comment:
suggestion: When combined with the autogenerated application resource documentation, this reads a bit awkwardly. 

```
Specify a resource other than the default for a charm. Note that not all charms have resources.

Notes:

 *  A resource can be specified by a revision number or by URL to a OCI image repository. Resources of type 'file' can only be specified by revision number. Resources of type 'oci-image' can be specified by revision number or URL to an OCI image repository.
 ```

Reply:
Explanation and formatting is updated.